contracts/satoshi-bridge/src/zcash_utils/orchard_policy.rs

```diff
@@ -0,0 +1,136 @@
+        network::Chain::ZcashTestnet => zcash_protocol::consensus::NetworkType::Test,
+        _ => return false, // Not a ZCash chain
+    };
+
+    let local_addr: network::Address = match zaddr.convert_if_network::<network::Address>(net) {
+        Ok(addr) => addr,
+        Err(_) => return false, // Network mismatch
+    };
+
+    // Check if it's a Unified Address
+    let ua = match local_addr {
+        network::Address::Unified { address, .. } => address,
+        _ => return false, // Not a Unified Address
+    };
+
+    // Check for Orchard receiver
+    for recv in ua.items_as_parsed() {
+        if let zcash_address::unified::Receiver::Orchard(_) = recv {
+            return true;
+        }
+    }
+    false
+}
+
+/// Extract the Orchard receiver raw bytes from a Unified Address string for the given chain.
+pub fn extract_orchard_receiver_from_unified(
+    target_addr: &str,
+    chain: &network::Chain,
+) -> [u8; 43] {
+    let zaddr = zcash_address::ZcashAddress::try_from_encoded(target_addr)
+        .expect("Invalid Zcash address encoding");
+
+    let net = match chain {
+        network::Chain::ZcashMainnet => zcash_protocol::consensus::NetworkType::Main,
+        network::Chain::ZcashTestnet => zcash_protocol::consensus::NetworkType::Test,
+        _ => near_sdk::env::panic_str("Unsupported chain for Orchard withdraw"),
+    };
+
+    let local_addr: network::Address = zaddr
+        .convert_if_network::<network::Address>(net)
+        .expect("Address network mismatch");
+
+    let ua = match local_addr {
+        network::Address::Unified { address, .. } => address,
+        _ => near_sdk::env::panic_str("Expected Unified Zcash address for Orchard withdraw"),
+    };
+
+    for recv in ua.items_as_parsed() {
+        if let zcash_address::unified::Receiver::Orchard(bytes) = recv {
+            return *bytes;
+        }
+    }
+    near_sdk::env::panic_str("Unified address missing Orchard receiver")
+}
+
+/// Validate Orchard bundle against policy:
+/// - Recovers all outputs using BRIDGE_OVK
+/// - Verifies exactly one non-zero output exists
+/// - Verifies the recovered amount is within expected range (allows dust adjustment)
+/// - Verifies the recovered recipient matches the expected Unified Address's Orchard receiver
+/// - Verifies value balance matches the output amount (value flows from transparent to Orchard)
+pub fn validate_orchard_bundle(
+    bundle: &Bundle<orchard::bundle::Authorized, ZatBalance>,
+    expected_recipient: &str,
+    expected_amount: u64,
+    min_change_amount: u64,
+    chain: &network::Chain,
+    orchard_output: (u64, [u8; 43]),
+) {
+    let (recovered_amount, recovered_addr_bytes) = orchard_output;
+
+    // Validate amount is within range (allow dust adjustment like transparent validation)
+    let expected_max = expected_amount;
+    let expected_min = expected_amount.saturating_sub(min_change_amount);
+
+    require!(
```

Comment:
Don't need this check

Reply:
fixed: https://github.com/Near-One/btc-bridge/pull/20/commits/7e64d5af5f9f1700d294a1a4a6fb83ce9a8335dd